### DISCUSSION TITLE
QRZ Integration not working Windows 2.6.16. Error in encryption.php

### DISCUSSION THREAD

**Discussion by dalekhog** [magicbug/Cloudlog#3206] [Q&A] [answered]
Hi, thanx for this excellent logging app.
SInce this version, 2.6.16, (full new install in another pc, Windows), when adding a QSO, fields are not populated.
Added qrz with credentials in Third Party Services, and also in the config.php file and still doesnt work.
If I select NONE in Third Party Services, fileds are populated with callsigns I have already worked.

Does anybody have this working?

Should I reinstall everything?

Thank you again,

Pablo
CX4RL

**Comment by magicbug** [comment]
After adding the logins are you logging out and back in again? that might solve it

**Reply by dalekhog** [reply]
HI, and thanx for answering so promptly.
I added in config.php qrz and in the part of qrz credentials also.
And also in the Third Party services.
Logged out and in several times. Didnt solve it.
I even restarted WAMP server serveral times and it doesnt solve it.
I have 2 FA in QRZ. I don't think that could be the problem because it worked before, and using HAMQTH doesn't work either.

Any ideas?

Can't thank you enough.
PAblo
CX4RL

**Reply by magicbug** [reply]
I wouldn't expect WAMP to be the issue, but might be worth turning on log reporting in /application/config/config.php and setting Logging to log all you'll find logs in /application/log and if you try type a callsign in that isn't in your log then check the log to see if it complains about anything

**Reply by dalekhog** [reply]
Hi again- I have confiured logging threshold to 4 (All messages).
The log file does not log any error when typing a new callsign.
Nothing gets populated, using qrz or hamqth.
If I set Third Party Services to NONE, it populates fields if callsign is in the log. Nothing if not, and says that lookup is not cofnigured.
Log with no errors.

Strange.

Thank you.
Pablo

**Reply by dalekhog** [reply]
Hi again.
Im getting these errors in the log when I input a callsign that I have not worked, with  qrz configured both in config.php and third pty services.
Please help....I love cloudlog, and I cannot use it without lookup,

Is it working for someone?

This is what Im getting

INFO - 2024-10-17 23:27:52 --> Database Driver Class Initialized
INFO - 2024-10-17 23:27:52 --> Session: Class initialized using 'files' driver.
INFO - 2024-10-17 23:27:52 --> Migrations Class Initialized
INFO - 2024-10-17 23:27:52 --> Language file loaded: language/english/migration_lang.php
INFO - 2024-10-17 23:27:52 --> Database Forge Class Initialized
INFO - 2024-10-17 23:27:52 --> Model "Options_model" initialized
INFO - 2024-10-17 23:27:52 --> Model "User_options_model" initialized
INFO - 2024-10-17 23:27:52 --> Controller Class Initialized
INFO - 2024-10-17 23:27:52 --> Model "User_Model" initialized
INFO - 2024-10-17 23:27:52 --> Model "Logbook_model" initialized
INFO - 2024-10-17 23:27:52 --> Model "Logbooks_model" initialized
INFO - 2024-10-17 23:27:52 --> File loaded: C:\wamp64\www\Cloudlog\application\views\qso/components/previous_contacts.php
INFO - 2024-10-17 23:27:52 --> Final output sent to browser
INFO - 2024-10-17 23:27:52 --> Total execution time: 0.2584
ERROR - 2024-10-17 23:29:05 --> Severity: 8192 --> strlen(): Passing null to parameter #1 ($string) of type string is deprecated C:\wamp64\www\Cloudlog\system\libraries\Encryption.php 914
ERROR - 2024-10-17 23:29:05 --> Severity: 8192 --> strlen(): Passing null to parameter #1 ($string) of type string is deprecated C:\wamp64\www\Cloudlog\system\libraries\Encryption.php 914
ERROR - 2024-10-17 23:30:59 --> Severity: 8192 --> strlen(): Passing null to parameter #1 ($string) of type string is deprecated C:\wamp64\www\Cloudlog\system\libraries\Encryption.php 914
ERROR - 2024-10-17 23:30:59 --> Severity: 8192 --> strlen(): Passing null to parameter #1 ($string) of type string is deprecated C:\wamp64\www\Cloudlog\system\libraries\Encryption.php 914


And got DEBUG ones: 
INFO - 2024-10-17 23:16:37 --> Config Class Initialized
INFO - 2024-10-17 23:16:37 --> Hooks Class Initialized
INFO - 2024-10-17 23:16:37 --> UTF-8 Support Enabled
INFO - 2024-10-17 23:16:37 --> Utf8 Class Initialized
INFO - 2024-10-17 23:16:37 --> URI Class Initialized
DEBUG - 2024-10-17 23:16:37 --> No URI present. Default controller set.

DEBUG - 2024-10-17 23:16:37 --> Migration class already loaded. Second attempt ignored.
DEBUG - 2024-10-17 23:16:37 --> Encryption: Auto-configured driver 'openssl'.

Thank you

Pablo

**Reply by magicbug** [reply]
Null parameter suggests no password was set 

What php version are you running?

I'll also add a test button to make sure this info is being correctly added

**Reply by dalekhog** [reply]
Hi and thank you.
I have the password set for qrz in 3rd party services and config.php.
Is that password you are referring to?

PhP version 8.2.18

No password set in Php my Admin.

This is what wampserver says:

Server Configuration
Apache Version:
2.4.59  - [Documentation Apache](http://httpd.apache.org/docs/2.4/en/)
Server Software:
Apache/2.4.59 (Win64) PHP/8.2.18 mod_fcgid/2.3.10-dev - Port defined for Apache: 80
PHP Version:
[Apache module]  8.2.18 - [Documentation PHP](http://www.php.net/manual/en/) - [Loaded PHP extensions](http://localhost/#phpextloaded) - [Use of PHP versions](http://localhost/#phpversionsuse)
 
[FCGI] 7.4.33 - 8.0.30 - 8.1.28 - 8.2.18 - 8.3.6 - [FCGI mode help](http://localhost/#helpfcgi)
MySQL Version:
8.3.0 - Port defined for MySQL: 3306 - default DBMS -  [Documentation MySQL](http://dev.mysql.com/doc/index.html)
MariaDB Version:
11.3.2 - Port defined for MariaDB: 3307 -  [Documentation MariaDB](http://mariadb.com/kb/en/mariadb/documentation) - [MySQL - MariaDB](http://localhost/#mysqlmariadb)

Thank you again.

Pablo

**Reply by magicbug** [reply]
Yep thats the information I wanted

**Reply by dalekhog** [reply]
Excellent.  Do you see something wrong? Any ideas?

Thanx

Pablo

**Reply by magicbug** [reply]
This is very likely to be a PHP version issue, while there is a bit of compatibility with PHP 8 the core of Cloudlog now is pretty old although stable.. I'm not sure if you can change PHP to 7.4 or not but worth a shot to use the recommended

**Reply by dalekhog** [reply]
Thank you.

I will change php to 7.4 and tell you what happens.

Pablo.-

**Comment by dalekhog** [comment, marked as answer]
That did it!!!

Just went to wamp server, php, version, clicked on version 7.4.32 that was there to select. Wamp serevr restarted, and voila! All is working!

Thank you thank you!

**Comment by magicbug** [comment]
CQWW SSB got in the way last week so only got around to sorting.

I've now committed a new patch to the master branch, the logout dance isn't required now and it seems to work as I'd expect, I'd suggest makings sure you are running the recommended PHP versions too. but if you have been having issues, update the code, re-enter the details

**Reply by dalekhog** [reply]
Many thanks. 
My problem was solved automatically by selecting version 7.4.32 of php in wamp server. 

Thank you for this excellent logging program.

Pablo 
CX4RL
